addons/delivery/models/sale_order.py

```diff
@@ -37,6 +37,14 @@ def onchange_order_line(self):
         if delivery_line:
             self.recompute_delivery_price = True
 
+    def update_prices(self):
+        """Recompute delivery after pricelist prices reset."""
+        super().update_prices()
+        if self.delivery_set:
```

Comment:
@wtaferner 
idea : set `recompute_delivery_price` to True
Like that is the user change the price list, the button "update delivery price" is show (in orange).

Reply:
Not sure if this what we want here as this would mean that the user might needs to update manually again what we just avoided by just resetting the correct price after `update_prices`.

Reply:
I think it is better in stable.